public/js/components/Graph.jsx

```diff
@@ -1,24 +1,24 @@
 import React from 'react'
+import { render } from 'react-dom'
 import { Chart } from 'react-google-charts'
+import GraphStore from '../stores/GraphStore'
+import GraphActions from '../actions/GraphActions'
 
 export default class Graph extends React.Component {
   constructor(props) {
     super(props)
     this.state = {
-      dataPoints: [],
+      dataPoints: GraphStore.getDataPoints(),
     }
   }
 
   componentDidMount() {
-    fetch('keywords/top')
-      .then(res => res.json())
-      .then((words) => {
-        const dataPoints = words[0].votes
-          .map(point => [point.time, point.sum])
-        dataPoints.unshift(['Time', words[0].word])
-        this.setState({ dataPoints })
+    GraphStore.on('graphUpdate', () => {
```

Comment:
Yes, you need to stop listening when you unmount. Also you need a separate function because you need to reference what listener (The function) to remove when you remove the listener on the event.